### ISSUE TITLE
Websocket is disconnecting when frequently sending messages

### ISSUE THREAD

**Issue by zekageri** [hoeken/PsychicHttp#19]

**Comment by zekageri** [comment]
Will check with `-DCORE_DEBUG_LEVEL=5`

**Comment by zekageri** [comment]
oh it does not compile with level 5 debug

```cpp
In file included from C:/Users/Pc/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:80,
                 from C:/Users/Pc/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:36,
                 from lib/PsychicHttp-master/src/PsychicCore.h:23,     
                 from lib/PsychicHttp-master/src/PsychicWebSocket.h:4, 
                 from lib/PsychicHttp-master/src/PsychicWebSocket.cpp:1:
lib/PsychicHttp-master/src/PsychicWebSocket.cpp: In member function 'virtual esp_err_t PsychicWebSocketHandler::handleRequest(PsychicRequest*)':
lib/PsychicHttp-master/src/PsychicWebSocket.cpp:204:80: error: 'req' was not declared in this scope
   ESP_LOGI(PH_TAG, "ws_handler: httpd_handle_t=%p, sockfd=%d, client_info:%d", req->handle,

         ^~~
C:/Users/Pc/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-log.h:127:72: note: in definition of macro 'log_i'        
 #define log_i(format, ...) log_printf(ARDUHAL_LOG_FORMAT(I, format), ##__VA_ARGS__)

 ^~~~~~~~~~~
lib/PsychicHttp-master/src/PsychicWebSocket.cpp:204:3: note: in expansion of macro 'ESP_LOGI'
   ESP_LOGI(PH_TAG, "ws_handler: httpd_handle_t=%p, sockfd=%d, client_info:%d", req->handle,
   ^~~~~~~~
lib/PsychicHttp-master/src/PsychicWebSocket.cpp:204:80: note: suggested alternative: 'ret'
   ESP_LOGI(PH_TAG, "ws_handler: httpd_handle_t=%p, sockfd=%d, client_info:%d", req->handle,

         ^~~
C:/Users/Pc/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-log.h:127:72: note: in definition of macro 'log_i'        
 #define log_i(format, ...) log_printf(ARDUHAL_LOG_FORMAT(I, format), ##__VA_ARGS__)

 ^~~~~~~~~~~
lib/PsychicHttp-master/src/PsychicWebSocket.cpp:204:3: note: in expansion of macro 'ESP_LOGI'
   ESP_LOGI(PH_TAG, "ws_handler: httpd_handle_t=%p, sockfd=%d, client_info:%d", req->handle,
   ^~~~~~~~
Compiling .pio\build\esp-wrover-kit\libb41\SD\SD.cpp.o
*** [.pio\build\esp-wrover-kit\lib14e\PsychicHttp-master\PsychicWebSocket.cpp.o] Error 1
==================== [FAILED] Took 126.74 seconds ====================
```

**Comment by zekageri** [comment]
I have commented out this line

```cpp
ESP_LOGI(PH_TAG, "ws_handler: httpd_handle_t=%p, sockfd=%d, client_info:%d", req->handle,
    httpd_req_to_sockfd(req), httpd_ws_get_fd_info(req->handle, httpd_req_to_sockfd(req)));
```

in `PsychicWebSocket.cpp` so it can compile. All i see is a dc

```cpp
[Server] - Connection #55 connected from 192.168.10.105
[ 38947][I][PsychicClient.cpp:45] localIP(): [psychic] Client Local IP 
=> 192.168.10.105
[HsH_Socket] - Connection #55 connected from 192.168.10.105
[ 38959][I][PsychicWebSocket.cpp:175] handleRequest(): [psychic] frame 
len is 46
[ 38959][I][PsychicWebSocket.cpp:191] handleRequest(): [psychic] Got packet with message: {"header":{"event":"test"},"data":{"test":29}}
[HsH_Socket] - Received: {"header":{"event":"test"},"data":{"test":29}}[ 38980][E][PsychicWebSocket.cpp:203] handleRequest(): [psychic] httpd_ws_send_frame failed with ERROR
[ 38985][I][PsychicHttpServer.cpp:247] closeCallback(): [psychic] Client disconnected 55
[ 38993][I][PsychicClient.cpp:45] localIP(): [psychic] Client Local IP 
=> 192.168.10.105
[HsH_Socket] - Connection #55 closed from 192.168.10.105
[ 39549][I][PsychicHttpServer.cpp:221] openCallback(): [psychic] New client connected 55
[ 39549][I][PsychicClient.cpp:45] localIP(): [psychic] Client Local IP 
=> 192.168.10.105
[Server] - Connection #55 connected from 192.168.10.105
[ 39567][I][PsychicClient.cpp:45] localIP(): [psychic] Client Local IP 
=> 192.168.10.105
[HsH_Socket] - Connection #55 connected from 192.168.10.105
[ 39599][I][PsychicHttpServer.cpp:247] closeCallback(): [psychic] Client disconnected 52
[ 39600][I][PsychicClient.cpp:45] localIP(): [psychic] Client Local IP 
=> 192.168.10.105
[Event] - Connection #52 closed from 192.168.10.105
[ 39619][I][PsychicHttpServer.cpp:247] closeCallback(): [psychic] Client disconnected 55
[ 39619][E][PsychicClient.cpp:39] localIP(): [psychic] Error getting client IP
[HsH_Socket] - Connection #55 closed from 0.0.0.0
```

My client does not close the connection neither the server. I have no such logic

**Comment by zekageri** [comment]
Ah yes of course it was a silly mistake.

I returned true and not 0 or OK in the onFrameCb....

**Wrong**
```cpp
ws.onFrame([this](PsychicWebSocketRequest *request, httpd_ws_frame *frame) {
    frame->payload[frame->len] = '\0';
    std::string message(reinterpret_cast<char*>(frame->payload));
    messages.push_back(std::move(message));
    return true;
});
```

**Good**
```cpp
ws.onFrame([this](PsychicWebSocketRequest *request, httpd_ws_frame *frame) {
    frame->payload[frame->len] = '\0';
    std::string message(reinterpret_cast<char*>(frame->payload));
    messages.push_back(std::move(message));
    return OK; // Or 0 because it expects an esp_err_t just like everything else. -1 or 0.
});
```
